src/shell.js

```diff
@@ -111,9 +111,12 @@ if (ENVIRONMENT_IS_NODE) {
 
   Module['arguments'] = process['argv'].slice(2);
 
+  // With MODULARIZE == 1 wrapper function will be exported instead of Module itself and re-exporting it second time will cause issues
```

Comment:
i still don't fully understand this even with the comment, sorry.

why is the second time special? and why does it matter we export the wrapper function and not the Module? I don't see a connection here that is important, I guess...

Reply:
Without this patch `var m = require('module_name')` initially exports wrapper function `module.exports = wrapper_function`, and we need to call this wrapper function to obtain `Module` itself.
However, when `m()` is called it re-exports `module.exports = Module`.
Thus successive `require('module_name')` doesn't return wrapper function anymore, but instead returns plain `Module`.

Just try to run added test on current `incoming` branch.

Reply:
There are two `Module` variables. When you `require` it the first time you get the outer wrapper `Module`, and the second time you get the inner true `Module`.

Reply:
Oh, I see, so there are two separate places we export, one inside the module - in shell.js, which is disabled by this PR - and another outside the module, added by MODULARIZE, and is the proper place?

If that's correct then looks good to me. Let's add that to the comment, though, something like `MODULARIZE will export the module in the proper place outside, we don't need to export here`.

Reply:
That is correct, updated comment accordingly